### ISSUE TITLE
error: item does not constrain `ShaderPassMeta::{opaque#0}`, but has it in its signature

### ISSUE THREAD

**Issue by Azq2** [SnowflakePowered/librashader#85]
Latest nightly rust:
```
   Compiling wgpu-core v0.19.4
   Compiling librashader-runtime-vk v0.2.7
   Compiling librashader-runtime-gl v0.2.7
error: item does not constrain `ShaderPassMeta::{opaque#0}`, but has it in its signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-gl-0.2.7/src/filter_chain/filter_impl.rs:194:8
    |
194 |     fn init_passes(
    |        ^^^^^^^^^^^
    |
    = note: consider moving the opaque type's declaration and defining uses into a separate module
note: this opaque type is in the signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-gl-0.2.7/src/filter_chain/filter_impl.rs:96:24
    |
96  |     ShaderPassArtifact<impl CompileReflectShader<GLSL, SpirvCompilation, SpirvCross>>;
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: item does not constrain `ShaderPassMeta::{opaque#0}`, but has it in its signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-gl-0.2.7/src/filter_chain/filter_impl.rs:217:51
    |
217 |                 reflection.ubo.as_ref().map_or(0, |ubo| ubo.size as usize),
    |                                                   ^^^^^
    |
    = note: consider moving the opaque type's declaration and defining uses into a separate module
note: this opaque type is in the signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-gl-0.2.7/src/filter_chain/filter_impl.rs:96:24
    |
96  |     ShaderPassArtifact<impl CompileReflectShader<GLSL, SpirvCompilation, SpirvCross>>;
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: item does not constrain `ShaderPassMeta::{opaque#0}`, but has it in its signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-gl-0.2.7/src/filter_chain/filter_impl.rs:221:32
    |
221 |                     .map_or(0, |push| push.size as usize),
    |                                ^^^^^^
    |
    = note: consider moving the opaque type's declaration and defining uses into a separate module
note: this opaque type is in the signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-gl-0.2.7/src/filter_chain/filter_impl.rs:96:24
    |
96  |     ShaderPassArtifact<impl CompileReflectShader<GLSL, SpirvCompilation, SpirvCross>>;
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: item does not constrain `ShaderPassMeta::{opaque#0}`, but has it in its signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-gl-0.2.7/src/filter_chain/filter_impl.rs:224:71
    |
224 |             let uniform_bindings = reflection.meta.create_binding_map(|param| {
    |                                                                       ^^^^^^^
    |
    = note: consider moving the opaque type's declaration and defining uses into a separate module
note: this opaque type is in the signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-gl-0.2.7/src/filter_chain/filter_impl.rs:96:24
    |
96  |     ShaderPassArtifact<impl CompileReflectShader<GLSL, SpirvCompilation, SpirvCross>>;
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: item does not constrain `ShaderPassMeta::{opaque#0}`, but has it in its signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-vk-0.2.7/src/filter_chain.rs:409:8
    |
409 |     fn init_passes(
    |        ^^^^^^^^^^^
    |
    = note: consider moving the opaque type's declaration and defining uses into a separate module
note: this opaque type is in the signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-vk-0.2.7/src/filter_chain.rs:213:24
    |
213 |     ShaderPassArtifact<impl CompileReflectShader<SPIRV, SpirvCompilation, SpirvCross> + Send>;
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: item does not constrain `ShaderPassMeta::{opaque#0}`, but has it in its signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-vk-0.2.7/src/filter_chain.rs:422:18
    |
422 |             .map(|(index, (config, source, mut reflect))| {
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: consider moving the opaque type's declaration and defining uses into a separate module
note: this opaque type is in the signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-vk-0.2.7/src/filter_chain.rs:213:24
    |
213 |     ShaderPassArtifact<impl CompileReflectShader<SPIRV, SpirvCompilation, SpirvCross> + Send>;
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: item does not constrain `ShaderPassMeta::{opaque#0}`, but has it in its signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-vk-0.2.7/src/filter_chain.rs:426:66
    |
426 |                 let ubo_size = reflection.ubo.as_ref().map_or(0, |ubo| ubo.size as usize);
    |                                                                  ^^^^^
    |
    = note: consider moving the opaque type's declaration and defining uses into a separate module
note: this opaque type is in the signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-vk-0.2.7/src/filter_chain.rs:213:24
    |
213 |     ShaderPassArtifact<impl CompileReflectShader<SPIRV, SpirvCompilation, SpirvCross> + Send>;
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: item does not constrain `ShaderPassMeta::{opaque#0}`, but has it in its signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-vk-0.2.7/src/filter_chain.rs:437:36
    |
437 |                         .map_or(0, |push| push.size as usize),
    |                                    ^^^^^^
    |
    = note: consider moving the opaque type's declaration and defining uses into a separate module
note: this opaque type is in the signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-vk-0.2.7/src/filter_chain.rs:213:24
    |
213 |     ShaderPassArtifact<impl CompileReflectShader<SPIRV, SpirvCompilation, SpirvCross> + Send>;
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: item does not constrain `ShaderPassMeta::{opaque#0}`, but has it in its signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-vk-0.2.7/src/filter_chain.rs:440:75
    |
440 |                 let uniform_bindings = reflection.meta.create_binding_map(|param| param.offset());
    |                                                                           ^^^^^^^
    |
    = note: consider moving the opaque type's declaration and defining uses into a separate module
note: this opaque type is in the signature
   --> /home/azq2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/librashader-runtime-vk-0.2.7/src/filter_chain.rs:213:24
    |
213 |     ShaderPassArtifact<impl CompileReflectShader<SPIRV, SpirvCompilation, SpirvCross> + Send>;
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `librashader-runtime-gl` (lib) due to 4 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `librashader-runtime-vk` (lib) due to 5 previous errors

```

**Comment by Azq2** [comment]
Workaround
```toml
[dependencies]
librashader = { git = "https://github.com/SnowflakePowered/librashader" }
```

**Comment by chyyran** [comment]
Will be pushing a release soon to fix this.

**Comment by chyyran** [comment]
Fixed in 0.2.8
